test/Dialect/SystemC/module-error.mlir

```diff
@@ -35,5 +35,50 @@
 
 // -----
 
+// expected-error @+1 {{module port names must be unique}}
+"systemc.module"() ({
+  ^bb0(%arg0: i4, %arg1: i32, %arg2: i4, %arg3: i8):
+  }) {function_type = (i4, i32, i4, i8) -> (), portDirections = #systemc.port_directions<[sc_out, sc_in, sc_out, sc_inout]>, portNames = ["port0", "port1", "port2", "port2"], sym_name = "verifierTest"} : () -> ()
+
+// -----
+
 // expected-error @+1 {{expected 'sc_in', 'sc_out', or 'sc_inout'}}
 systemc.module @parserTest (sc_invalid %arg: i32) { }
+
+// -----
+
+// expected-error @+1 {{declared names must be unique}}
+systemc.module @signalNameConflict () {
+  %0 = "systemc.signal"() {name = "signal0"} : () -> i32
+  %1 = "systemc.signal"() {name = "signal0"} : () -> i32
+}
+
+// -----
+
+// expected-error @+1 {{declared names must be unique}}
+systemc.module @signalNameConflictWithArg (sc_in %in: i32) {
+  %0 = "systemc.signal"() {name = "in"} : () -> i32
+}
```

Comment:
This type of test would motivate additional error reporting that the signal name conflicts with the port name. The general form of this is probably changing the reporting to state: 
1. An error on the SCModule (not printing the full module).
2. Iterating over all the conflicts and printing the problematic ops/ports.

Reply:
I changed your suggestion slightly to report the error on the op that redefines the name and add notes to the first definition of the name as well as the module the conflict occurs in.